python_modules/libraries/dagster-aws/dagster_aws/s3/sensor.py

```diff
@@ -22,19 +43,23 @@ def get_s3_keys(
         raise botocore.exceptions.ClientError("Failed to initialize s3 client")
 
     paginator = s3_session.get_paginator("list_objects_v2")
-    page_iterator = paginator.paginate(Bucket=bucket, Prefix=prefix)
+    page_iterator = paginator.paginate(Bucket=bucket, Prefix=prefix, Delimiter="")
 
     contents = []
     for page in page_iterator:
         contents.extend(page.get("Contents", []))
 
-    sorted_keys = [obj["Key"] for obj in sorted(contents, key=lambda x: x["LastModified"])]
+    sorted_objs = [obj for obj in sorted(contents, key=lambda x: x["LastModified"])]
 
-    if not since_key or since_key not in sorted_keys:
-        return sorted_keys
+    if since_last_modified:
+        since_last_modified_dt = datetime.fromisoformat(since_last_modified)
+        for idx, obj in enumerate(sorted_objs):
+            if obj.get("LastModified") >= since_last_modified_dt:
+                return [obj.get("Key") for obj in sorted_objs[idx + 1 :]]
```

Comment:
Hey @maximearmstrong and @danielgafni - I was hoping to get your input on this as practitioners.

The context of the issue can be found here:
https://github.com/dagster-io/dagster/issues/22892

I wanted to gauge your input on if you think it makes sense to handle these two scenarios in the same function: getting objects since a specific s3 key, and getting objects since a last modified timestamp (which was added in this PR). Or would it make sense to separate these functions?

The current limitation of this approach, is that we are not returning the Last Modified timestamp in this function, so if the user does not have a way to store that on the context of their sensor for the next run. Instead, they could only determine the date from some external state (like when the job has run).

Should we create a separate function that returns a list of tuples containing the key and the timestamp? Or is there some other approach that might be more ideal? Thanks!

Reply:
I think we should add another function which returns more information about the object. 

It can ever return the full information present in `s3.head_object`.

Reply:
Good call, updated. Thanks!